Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/TempSpawnableAssetsCache.cpp

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/Prefab/Spawnable/TempSpawnableAssetsCache.h>
+
+#include <AzCore/Asset/AssetManager.h>
+#include <AzCore/Asset/AssetManagerBus.h>
+#include <AzToolsFramework/Prefab/PrefabLoader.h>
+#include <AzToolsFramework/Prefab/PrefabSystemComponentInterface.h>
+
+namespace AzToolsFramework::Prefab::PrefabConversionUtils
+{
+    TempSpawnableAssetsCache::~TempSpawnableAssetsCache()
+    {
+        Deactivate();
+    }
+
+    bool TempSpawnableAssetsCache::Activate(AZStd::string_view stackProfile)
+    {
+        AZ_Assert(!IsActivated(), "TempSpawnableAssetsCache - TempSpawnableAssetsCache is currently activated");
+
+        m_prefabSystemComponentInterface = AZ::Interface<PrefabSystemComponentInterface>::Get();
+        AZ_Assert(m_prefabSystemComponentInterface, "TempSpawnableAssetsCache - Could not retrieve instance of PrefabSystemComponentInterface");
+
+        m_loaderInterface = AZ::Interface<Prefab::PrefabLoaderInterface>::Get();
+        AZ_Assert(m_loaderInterface, "TempSpawnableAssetsCache - Could not retrieve instance of PrefabLoaderInterface");
+
+        m_stockProfile = stackProfile;
+        return m_converter.LoadStackProfile(m_stockProfile);
+    }
+
+    void TempSpawnableAssetsCache::Deactivate()
+    {
+        ClearAllTempSpawnableRelatedAssets();
+        m_stockProfile = "";
+        m_loaderInterface = nullptr;
+        m_prefabSystemComponentInterface = nullptr;
+    }
+
+    bool TempSpawnableAssetsCache::IsActivated() const
+    {
+        return m_converter.IsLoaded();
+    }
+
+    AZStd::string_view TempSpawnableAssetsCache::GetStockProfile() const
+    {
+        return m_stockProfile;
+    }
+
+    TempSpawnableAssetsCache::CreateSpawnableResult TempSpawnableAssetsCache::CreateTempSpawnableAsset(
```

Comment:
"Temp" is n indication on how the class is used not what it does. A better name imho is "in memory".
As a recommendation for alternative class and function names, consider "InMemorySpawnableAssetContainer" and "CreateInMemorySpawnableAsset".